server/test/unit/vitalsTest/checkForInternalProblemsTest.js

```diff
@@ -0,0 +1,215 @@
+      await vitals.checkForInternalProblems()
+    })
+
+    it('should not send any alerts to Sentry', () => {
+      expect(helpers.logSentry).to.not.have.been.called
+    })
+
+    it('should not log any errors', () => {
+      expect(helpers.log).to.not.be.called
+    })
+  })
+
+  describe('for a location that has no previous sessions', () => {
+    beforeEach(async () => {
+      // Insert a single location with no sessions
+      const client = await factories.clientDBFactory(db, { isActive: true })
+      this.testLocation = await locationDBFactory(db, {
+        clientId: client.id,
+        isActive: true,
+      })
+      await vitals.checkForInternalProblems()
+    })
+
+    it('should not send any alerts to Sentry', () => {
+      expect(helpers.logSentry).to.not.have.been.called
+    })
+
+    it('should not log any errors', () => {
+      expect(helpers.log).to.not.be.called
+    })
+  })
+
+  describe('should not log to Sentry if exactly maxStillnessAlerts have occurred in 24 hours at a location', () => {
+    beforeEach(async () => {
+      // Insert a single location and maxStillnessAlerts sessions + 1
+      const client = await factories.clientDBFactory(db, { isActive: true })
+      this.testLocation = await locationDBFactory(db, {
+        clientId: client.id,
+        isActive: true,
+      })
+
+      this.incidentCategory = ALERT_TYPE.SENSOR_STILLNESS
+      this.alertType = ALERT_TYPE.SENSOR_STILLNESS
+      for (let i = 1; i <= maxStillnessAlerts; i++) {
+        this[`session${i}`] = await sessionDBFactory(db, {
+          locationid: this.testLocation.locationid,
+          alertType: this.alertType,
+          incidentCategory: this.incidentCategory,
+        })
+        await helpers.sleep(1) // sleep to make sure that these have created_at values at least one millisecond apart
+      }
+      await vitals.checkForInternalProblems()
+    })
+
+    it('should not send any alerts to Sentry', () => {
+      expect(helpers.logSentry).to.not.have.been.called
+    })
+
+    it('should not log any errors', () => {
+      expect(helpers.log).to.not.be.called
+    })
+  })
+
+  describe('duration alerts should not be counted, even if duration alerts exceed maxStillnessAlerts in 24 hours for a location', () => {
+    beforeEach(async () => {
+      // Insert a single location and maxStillnessAlerts sessions + 1
+      const client = await factories.clientDBFactory(db, { isActive: true })
+      this.testLocation = await locationDBFactory(db, {
+        clientId: client.id,
+        isActive: true,
+      })
+      for (let i = 1; i <= maxStillnessAlerts + 1; i++) {
+        this[`session${i}`] = await sessionDBFactory(db, {
+          locationid: this.testLocation.locationid,
+          alertType: ALERT_TYPE.SENSOR_DURATION,
+          incidentCategory: ALERT_TYPE.SENSOR_DURATION,
+        })
+        await helpers.sleep(1) // sleep to make sure that these have created_at values at least one millisecond apart
```

Comment:
Did you run into issues that this call to `helpers.sleep` fix?

Reply:
I assumed that I would when I put that in, but I didn't run into any issues taking it out! I think I left it in to be safe!

Reply:
Actually I think it doesn't matter hahaha I'll go ahead and remove it!